### PR TITLE
fix(segment-button): apply hover properly for segment with color

### PR DESCRIPTION
**Pull request by brandyscarney**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [ ] Build (`npm run build`) was run locally and any changes were pushed
- [ ] Lint (`npm run lint`) has passed locally and any fixes were made for failures

## Pull request type

 

 

Please check the type of change your PR introduces:
- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?

Issue Number: N/A

Can be reproduced the following way:

1. Go to http://localhost:3333/src/components/segment/test/basic
1. Hover over any segments with a color (either mode) and notice that there is no hover

_or_

1. Go to http://localhost:3333/src/themes/test/css-variables
1. Click the colors tab and try to hover the segments, notice there is no hover

## What is the new behavior?

The hover state should be fixed, in both modes, on my branch

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: core/src/components/segment/segment.md.scss [hunk 1]
```diff
@@ -40,6 +39,11 @@
   color: #{current-color(base)};
 }
 
+@media (any-hover: hover) {
+  :host(.ion-color)::slotted(:hover:not(.segment-button-checked)) {
```
Comment (liamdebeasi) [inline core/src/components/segment/segment.md.scss]: This doesn't seem to be working for selected segments w/ color.

Comment (brandyscarney) [reply core/src/components/segment/segment.md.scss]: Yeah that has been intentional, I'm sure if hover should work on one that is already checked.

Comment (liamdebeasi) [reply core/src/components/segment/segment.md.scss]: Hmm interesting. It looks like the hover is working for selected segments that just default to the primary color.

Comment (brandyscarney) [reply core/src/components/segment/segment.md.scss]: Oh sorry, I thought this was iOS. I see it on MD.

**Comment by brandyscarney** [comment]
@liamdebeasi Do you want to re-review this? The changes were made

**Pull request merged**